debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java

```diff
@@ -195,14 +205,40 @@ protected void readTableStructure(ChangeEventSourceContext sourceContext, Relati
                 throw new InterruptedException("Interrupted while reading structure of schema " + schema);
             }
 
+            // todo: DBZ-137 the new readSchemaForCapturedTables seems to cause failures.
+            // For now, reverted to the default readSchema implementation as the intended goal
+            // with the new implementation was to be faster, not change behavior.
+            // if (connectorConfig.getAdapter().equals(OracleConnectorConfig.ConnectorAdapter.LOG_MINER)) {
+            // jdbcConnection.readSchemaForCapturedTables(
+            // snapshotContext.tables,
+            // snapshotContext.catalogName,
+            // schema,
+            // connectorConfig.getColumnFilter(),
+            // false,
+            // snapshotContext.capturedTables);
+            // }
+            // else {
             jdbcConnection.readSchema(
                     snapshotContext.tables,
                     snapshotContext.catalogName,
                     schema,
                     connectorConfig.getTableFilters().dataCollectionFilter(),
                     null,
                     false);
+            // }
+        }
+    }
+
+    @Override
```

Comment:
I'd remove that here and revisit this more holistically in https://issues.redhat.com/browse/DBZ-2525.

Reply:
I've removed the pertinent code here and I'll re-add it as part of DBZ-2525.